### ISSUE TITLE
BIP-85 derivation algorithm: unnecessary index limitation

### ISSUE THREAD

**Issue by AnK26-616** [airgap-it/airgap-vault#217]
BIP85 based derivation algorithm is very useful not only for storing / deriving child wallet backup phrases but also for other solutions when the 12-24 word list can be used as a password to secure some secret. However the AirGap Vault (as of version 3.32.4) allows only to choose from the first 10 indexes (0-9). That is unnecessary limitation. It would be super useful if one could enter the index number simply by hand and then the app would derive the proper child mnemonic under given index. Is it possible to add that feature, please?

**Comment by AndreasGassmann** [comment]
This limitation was added to prevent accidental misuse. Now that we have the advanced mode, I don't see any problem with allowing any kind of index.

**Comment by AnK26-616** [comment]
You mean adding it under advance mode? That would be really great! As currently advanced mode allows only BIP-39 passphrases which is different thing than index (tbh way more important than index - but the compatibility between different tools is something I am trying to address with Air Gap Vault)

**Comment by AndreasGassmann** [comment]
Yes. We can make it so that if you have advanced mode enabled, any kind of index can be used. Should we allow any index, or up to `2147483647`?

**Comment by AnK26-616** [comment]
Good question. Well... I am not sure if the effort to implement "any" really is worth spending. I would consider 32 bit integer at least - possibly with 64 bit integer limitation - 18446744073709551615- as the good "balance" between those two options (and also more future proof).

**Comment by AnK26-616** [comment]
I found that the "natural" limitation is the size of integer 2147483647. So that would be more than enough.

**Comment by allodial** [comment]
Yes please allow with upper limit 2147483647 in advanced mode.
